src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -697,10 +739,18 @@ def __write_status_file(self):
             shutil.rmtree(self.complete_status_file_path)
 
         # Write complete status file <seq.no>.complete.status
-        self.env_layer.file_system.write_with_retry_using_temp_file(self.complete_status_file_path, '[{0}]'.format(json.dumps(complete_status_payload)), mode='w+')
+        status_file_payload_json_dumps = json.dumps(complete_status_payload)
+        self.env_layer.file_system.write_with_retry_using_temp_file(self.complete_status_file_path, '[{0}]'.format(status_file_payload_json_dumps), mode='w+')
+
+        if Constants.StatusTruncationConfig.TURN_ON_TRUNCATION:
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(status_file_payload_json_dumps)     # calc complete_status_file_payload byte size on disk
 
-        # Write agent status file
-        self.env_layer.file_system.write_with_retry_using_temp_file(self.status_file_path, '[{0}]'.format(json.dumps(complete_status_payload)), mode='w+')
+            if status_file_size_in_bytes > self.__internal_file_capacity:  # perform truncation complete_status_file byte size > 126kb
+                truncated_status_file = self.__create_truncated_status_file(status_file_size_in_bytes, status_file_payload_json_dumps)
+                status_file_payload_json_dumps = json.dumps(truncated_status_file)
```

Comment:
Everything from line 746 to 750 should go in a function. i.e. if truncation is enabled, call a function that first checks if truncation is needed, and then applies truncation, and returns the truncated status.

Reply:
done